test/com/google/enterprise/adaptor/ad/AdAdaptorTest.java

```diff
@@ -1425,6 +1419,44 @@ public void testFakeAdaptorGetDocIds() throws Exception {
   }
 
   @Test
+  public void testMultipleFullGroupPushes() throws Exception {
+    MockLdapContext ldapContext = defaultMockLdapContext();
+    AdAdaptor adAdaptor = new FakeAdaptor(ldapContext);
+    RecordingDocIdPusher pusher = new RecordingDocIdPusher();
+    Map<String, String> configEntries = defaultConfig();
+    pushGroupDefinitions(adAdaptor, configEntries, pusher, /*fullPush=*/ true,
+        /*init=*/ true);
+
+    HashMap<GroupPrincipal, Collection<Principal>> goldenGroups =
+        new HashMap<GroupPrincipal, Collection<Principal>>();
+    Principal everyone = new GroupPrincipal("Everyone", "Default");
+    goldenGroups.put((GroupPrincipal) everyone, new ArrayList<Principal>());
+    goldenGroups.put(new GroupPrincipal("Authenticated Users@NT Authority",
+        "Default"), Arrays.asList(everyone));
+    goldenGroups.put(new GroupPrincipal("Interactive@NT Authority",
+        "Default"), Arrays.asList(everyone));
+    assertEquals(goldenGroups, pusher.getGroupDefinitions());
+
+    // Add the sam group.
+    addDefsToMockLdapContext(ldapContext, false);
+    pushGroupDefinitions(adAdaptor, configEntries, pusher, /*fullPush=*/ true,
+        /*init=*/ false);
+    @SuppressWarnings("unchecked")
+    HashMap<GroupPrincipal, Collection<Principal>> newGoldenGroups =
```

Comment:
I would prefer to indent this line four spaces. What's important about the clone that could not be accomplished with newGoldenGroups = new HashMap<>(goldenGroups)?

Reply:
Nothing special about clone. Started with Guava copyOf, but realized guava wasn't used here. Switched to the ctor that takes an existing map (just not the diamond operator version, since this adaptor uses -source=1.6).